File: boj/2023/plzrun/8-brute-230604/1182-dfs.cpp
#include <cstdio>

int N, S;
int A[20];

int dfs(int i, int diff)
{
	int ret = (diff == 0);
	for (int j = i; j < N; j++)
		ret += dfs(j + 1, diff - A[j]);
	return ret;	
}

int main()
{
	scanf("%d %d", &N, &S);
	for (int i = 0; i < N; i++)
		scanf("%d", &A[i]);
	printf("%d", dfs(0, S) - (S == 0));
}

File: boj/2023/plzrun/8-brute-230604/1182-dfs.py
import sys
input = lambda: sys.stdin.readline().rstrip()

N, S = map(int, input().split())
A = list(map(int, input().split()))

def dfs(i, diff):
	ret = (diff == 0)
	for j in range(i, len(A)):
		ret += dfs(j + 1, diff - A[j])
	return ret
print(dfs(0, S) - (S == 0))